### PR TITLE
build: do lustre available check only if --enable-lustre is set

### PR DESCRIPTION
**Pull request by KnightKu**
Signed-off-by: Gu Zheng <cengku@gmail.com>

### CODE REVIEW

File: m4/lx_find_lustre.m4 [hunk 1]
```diff
@@ -7,18 +7,20 @@ AC_DEFUN([X_AC_LUSTRE], [
...
-  AC_MSG_CHECKING([for Lustre])
-  AS_IF([test -e "/usr/include/lustre/lustre_user.h" && test -e "/usr/include/lustre/lustreapi.h"],
-        [enable_lustre="yes"],
-        [enable_lustre="no"])
-  AC_MSG_RESULT([$enable_lustre])
+  if [[ "x$enable_lustre" == xyes ]]; then
+    AC_MSG_CHECKING([for Lustre])
+    AS_IF([test -e "/usr/include/lustre/lustre_user.h" && test -e "/usr/include/lustre/lustreapi.h"],
+          [lustre_available="yes"],
+          [enable_available="no"])
```
Comment (dinatale2) [inline m4/lx_find_lustre.m4]: Should this line be `[lustre_available="no"])`?

Comment (KnightKu) [reply m4/lx_find_lustre.m4]: @Natale, thanks, it's typo, let me fix it and push again

**Comment by dinatale2** [comment]
@KnightKu Did you intend to append a bunch of extra commits to this PR?

**Comment by KnightKu** [comment]
Hi @Natale, please hold on a moment, too many patches are appended on this pull, will split them soon, and let's push one by one.

**Comment by dsikich** [comment]
@KnightKu Thanks! It makes it much easier for us to review that way.

**Pull request merged**